library_generation/utils/commit_message_formatter.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import List
+from typing import Dict
+from git import Commit
+
+
+def format_commit_message(commits: Dict[Commit, str], is_monorepo: bool) -> List[str]:
```

Comment:
nit: with a dict, you may not be guaranteed on the order that the commits are processed. it may not be a big deal for you, but something to consider if you potentially get inconsistent behavior for the "same" input.

Reply:
According to [dictionary doc](https://docs.python.org/3.11/library/typing.html), the order of items is insertion order by default.